internal/app/plan.go

```diff
@@ -156,28 +128,58 @@ func (p *plan) exec() {
 	}
 }
 
+func releaseWithHooks(cmd orderedCommand, wg *sync.WaitGroup, sem chan struct{}, errors chan string) {
+	defer func() {
+		wg.Done()
+		<-sem
+	}()
+	for _, c := range cmd.beforeCommands {
+		if msg, ok := execOne(c, cmd.targetRelease); !ok {
+			errors <- msg
+			log.Verbose(msg)
+			return
+		}
+	}
+	if msg, ok := execOne(cmd.Command, cmd.targetRelease); !ok {
+		errors <- msg
+		log.Verbose(msg)
+		return
+	}
+	if cmd.targetRelease != nil && !flags.dryRun && !flags.destroy {
+		cmd.targetRelease.label()
+	}
+	for _, c := range cmd.afterCommands {
+		if msg, ok := execOne(c, cmd.targetRelease); !ok {
+			errors <- msg
+			log.Verbose(msg)
+			return
+		}
+	}
+}
+
 // execOne executes a single ordered command
-func execOne(cmd command, targetRelease *release) {
+func execOne(cmd command, targetRelease *release) (msg string, ok bool) {
```

Comment:
why not just return an `error` instead of a `string` and a `bool`?

Reply:
oh, I keep on confusing approaches between languages, sorry, of course I changed that